### PR TITLE
hackgen-font: init at 2.7.1

### PR DESCRIPTION
**Pull request by natsukium**
###### Description of changes

HackGen is a Japanese composite font of Hack and GenJyuu-Gothic.
https://github.com/yuru7/HackGen

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/data/fonts/hackgen/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+  lib,
+  fetchzip,
+}: let
+  version = "2.7.1";
+in
```
Comment (AndersonTorres) [inline pkgs/data/fonts/hackgen/default.nix:6]:
```suggestion
{  lib
,  fetchzip
}: 

```

File: pkgs/data/fonts/hackgen/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  fetchzip {
+    name = "hackgen-font-${version}";
+    url = "https://github.com/yuru7/HackGen/releases/download/v${version}/HackGen_v${version}.zip";
+    postFetch = ''
+      install -Dm644 $out/*.ttf -t $out/share/fonts/hackgen
+      shopt -s extglob dotglob
+      rm -rf $out/!(share)
+      shopt -u extglob dotglob
+    '';
+    sha256 = "sha256-UL6U/q2u1UUP31lp0tEnFjzkn6dn8AY6hk5hJhPsOAE=";
```
Comment (AndersonTorres) [inline pkgs/data/fonts/hackgen/default.nix:16]:
```suggestion
```

File: pkgs/data/fonts/hackgen/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+  lib,
+  fetchzip,
+}: let
+  version = "2.7.1";
+in
+  fetchzip {
+    name = "hackgen-font-${version}";
+    url = "https://github.com/yuru7/HackGen/releases/download/v${version}/HackGen_v${version}.zip";
+    postFetch = ''
```
Comment (AndersonTorres) [inline pkgs/data/fonts/hackgen/default.nix:10]:
```suggestion
    url = "https://github.com/yuru7/HackGen/releases/download/v${version}/HackGen_v${version}.zip";
    sha256 = "sha256-UL6U/q2u1UUP31lp0tEnFjzkn6dn8AY6hk5hJhPsOAE=";
    
    postFetch = ''
```

File: pkgs/data/fonts/hackgen/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      install -Dm644 $out/*.ttf -t $out/share/fonts/hackgen
+      shopt -s extglob dotglob
+      rm -rf $out/!(share)
+      shopt -u extglob dotglob
+    '';
+    sha256 = "sha256-UL6U/q2u1UUP31lp0tEnFjzkn6dn8AY6hk5hJhPsOAE=";
+
+    meta = with lib; {
+      description = "composite font of Hack and GenJyuu-Goghic";
```
Comment (AndersonTorres) [inline pkgs/data/fonts/hackgen/default.nix:19]:
```suggestion
      description = "A composite font of Hack and GenJyuu-Goghic";
```

File: pkgs/data/fonts/hackgen/default.nix [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+      homepage = "https://github.com/yuru7/HackGen";
+      license = licenses.ofl;
+      platforms = platforms.unix;
+      maintainers = with maintainers; [natsukium];
```
Comment (AndersonTorres) [inline pkgs/data/fonts/hackgen/default.nix:23]:
```suggestion
      maintainers = with maintainers; [ natsukium ];
```

File: pkgs/data/fonts/hackgen/default.nix [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+      rm -rf $out/!(share)
+      shopt -u extglob dotglob
+    '';
+    sha256 = "sha256-UL6U/q2u1UUP31lp0tEnFjzkn6dn8AY6hk5hJhPsOAE=";
+
+    meta = with lib; {
+      description = "composite font of Hack and GenJyuu-Goghic";
+      homepage = "https://github.com/yuru7/HackGen";
+      license = licenses.ofl;
+      platforms = platforms.unix;
```
Comment (AndersonTorres) [inline pkgs/data/fonts/hackgen/default.nix:22]: Fonts are supposed to run everywhere.

```suggestion
```

File: pkgs/data/fonts/hackgen/nerdfont.nix [hunk 7]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (AndersonTorres) [inline pkgs/data/fonts/hackgen/nerdfont.nix:1]: Apply the changes here mutatis mutandis.

File: pkgs/data/fonts/hackgen/nerdfont.nix [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+    postFetch = ''
+      install -Dm644 $out/*.ttf -t $out/share/fonts/hackgen-nf
+      shopt -s extglob dotglob
+      rm -rf $out/!(share)
+      shopt -u extglob dotglob
+    '';
+    sha256 = "sha256-9sylGr37kKIGWgThZFqL2y6oI3t2z4kbXYk5DBMIb/g=";
+
+    meta = with lib; {
+      description = "composite font of Hack and GenJyuu-Gothic";
```
Comment (AndersonTorres) [inline pkgs/data/fonts/hackgen/nerdfont.nix:19]: Please provide some additional info; why this font differs?

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by natsukium** [comment]
@AndersonTorres 
Thank you for your review!
I have fixed the issues you pointed out.

By the way, I don't want to worry about where to break lines, where to put spaces, and so on, so I formatted using alejandra.
Is there any reference anywhere to nixpkgs' recommended code style?
For now, I format with alejandra -> nixpkgs-fmt in order.

**Comment by AndersonTorres** [comment]
> By the way, I don't want to worry about where to break lines, where to put spaces, and so on, so I formatted using alejandra.

OK.

> Is there any reference anywhere to nixpkgs' recommended code style?

Not yet. Maybe I need to write an RFC for that, but for now it is more about common sense instead of strict rules.

> For now, I format with alejandra -> nixpkgs-fmt in order.

I need to learn more about them.

File: pkgs/data/fonts/hackgen/default.nix [hunk 9]
```diff
@@ -0,0 +1,23 @@
+{ lib
+, fetchzip
+}:
```
Comment (AndersonTorres) [inline pkgs/data/fonts/hackgen/default.nix:3]: For now, add a blank line after this line here...

File: pkgs/data/fonts/hackgen/default.nix [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+fetchzip rec {
+  pname = "hackgen-font";
+  version = "2.7.1";
```
Comment (AndersonTorres) [inline pkgs/data/fonts/hackgen/default.nix:6]: And another after here.

Do the same on the other file.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
After that, squash the commits into two, one for each font you are adding.

**Comment by natsukium** [comment]
done!

**Pull request merged**